StripeCore/StripeCore/Source/DownloadManager/DownloadManager.swift

```diff
@@ -93,19 +93,34 @@ extension DownloadManager {
         let blockingDownloadSemaphore = DispatchSemaphore(value: 0)
 
         let urlRequest = URLRequest(url: url, cachePolicy: .useProtocolCachePolicy)
-        let task = self.session.downloadTask(with: url) { tempURL, response, _ in
-            guard let tempURL = tempURL,
-                let response = response,
-                let data = self.getDataFromURL(tempURL),
-                let image = self.persistToMemory(data, forImageName: imageName)
-            else {
-                let errorAnalytic = ErrorAnalytic(event: .unexpectedStripeCoreDownloadManagerError,
-                                                  error: Error.downloadSyncFailure,
+        let task = self.session.downloadTask(with: url) { tempURL, response, responseError in
+            if let responseError = responseError as? NSError {
+                blockingDownloadSemaphore.signal()
+                let errorAnalytic = ErrorAnalytic(event: .stripeCoreDownloadManagerError,
+                                                  error: responseError,
+                                                  additionalNonPIIParams: ["url": url.absoluteString])
+                STPAnalyticsClient.sharedClient.log(analytic: errorAnalytic)
+                return
+            }
+
+            guard let tempURL = tempURL, let response = response, let data = self.getDataFromURL(tempURL) else {
+                blockingDownloadSemaphore.signal()
+                let errorAnalytic = ErrorAnalytic(event: .stripeCoreDownloadManagerError,
+                                                  error: Error.failedToMakeDataFromResponse,
```

Comment:
The implementation of `getDataFromURL` hides its error from the caller, ideally we are able to report the error.

Reply:
Good point, after refactoring to use `data` on the URLSession we could remove the function entirely.